drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -204,25 +207,76 @@ static int mcux_flexcomm_transfer(const struct device *dev,
 static struct mcux_flexcomm_target_data *mcux_flexcomm_find_free_target(
 		struct mcux_flexcomm_data *data)
 {
-	struct mcux_flexcomm_target_data *target = &data->target_data;
+	int i;
+	struct mcux_flexcomm_target_data *target;
 
-	if (!target->target_attached) {
-		return target;
+	for (i = 0; i < ARRAY_SIZE(data->target_data); i++) {
+		target = &data->target_data[i];
+		if (!target->target_attached) {
+			return target;
+		}
 	}
 	return NULL;
 }
 
 static struct mcux_flexcomm_target_data *mcux_flexcomm_find_target_by_address(
 		struct mcux_flexcomm_data *data, uint16_t address)
 {
-	struct mcux_flexcomm_target_data *target = &data->target_data;
+	int i;
+	struct mcux_flexcomm_target_data *target;
 
-	if (target->target_attached && target->target_cfg->address == address) {
-		return target;
+	for (i = 0; i < ARRAY_SIZE(data->target_data); i++) {
+		target = &data->target_data[i];
+		if (target->target_attached && target->target_cfg->address == address) {
+			return target;
+		}
 	}
 	return NULL;
 }
 
+static int mcux_flexcomm_setup_i2c_config_address(struct mcux_flexcomm_data *data,
+		struct mcux_flexcomm_target_data *target, uint8_t address, bool disabled)
+{
+	i2c_slave_address_t *addr;
+	int idx = -1;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(data->target_data); i++) {
+		if (&data->target_data[i] == target) {
```

Comment:
NIT: you are not checking target_attached here

Reply:
Added check.